src/components/mdSwitch/mdSwitch.vue

```diff
@@ -66,9 +66,22 @@
         this.leftPos = this.checked ? checkedPosition + '%' : initialPosition;
       },
       changeState(checked, $event) {
-        this.checked = checked;
-        this.$emit('change', this.checked, $event);
-        this.$emit('input', this.checked, $event);
+        if ($event !== undefined) {
```

Comment:
If you want to test undefined, please test the type:
typeof $event !== 'undefined'